### PR TITLE
Base Update RPCs

### PR DESCRIPTION
**Pull request by 0xB10C**
This PR adds the counterparts to the UpdateBase and GetBaseUpdateProgress RPCs added in https://github.com/digitalbitbox/bitbox-base/pull/228 to the App. Next to the RPCs a event for a BaseUpdateProgressChanged notification is added.

### CODE REVIEW

File: backend/bitboxbase/rpcmessages/rpcmessages.go [hunk 1]
```diff
@@ -14,13 +14,15 @@ const (
...
+	// OpBaseUpdateProgressChanged notifies when the BaseUpdateProgress changes while performing a Base Update.
+	OpBaseUpdateProgressChanged = "u"
 )
 
 /*
 Put Incoming Args below this line. They should have the format of 'RPC Method Name' + 'Args'.
 */
 
-// UserAuthenticateArgs is a struct that holds the arguments for the UserAuthenticate RPC call. The first rpc call should always authenticate first
+// UserAuthenticateArgs is an struct that holds the arguments for the UserAuthenticate RPC call
```
Comment (Tomasvrba) [inline backend/bitboxbase/rpcmessages/rpcmessages.go]: Why change "a struct" to "an struct"? It should be the former

File: backend/bitboxbase/rpcmessages/rpcmessages.go [hunk 2]
```diff
@@ -94,7 +102,30 @@ type VerificationProgressResponse struct {
 	VerificationProgress float64 `json:"verificationProgress"`
 }
 
-// GetBaseInfoResponse is the struct that get sent by the rpc server during a GetBaseInfo rpc call
+// BaseUpdateState is the used to hold the current state for a Base update.
```
Comment (Tomasvrba) [inline backend/bitboxbase/rpcmessages/rpcmessages.go]: Missing noun. 
> BaseUpdateState is the [missing] used...

File: backend/bitboxbase/handlers/handlers.go [hunk 3]
```diff
@@ -433,6 +449,24 @@ func (handlers *Handlers) postRebootBaseHandler(_ *http.Request) (interface{}, e
 	}, nil
 }
 
+func (handlers *Handlers) postUpdateBaseHandler(r *http.Request) (interface{}, error) {
+	handlers.log.Debug("Update Base")
+	payload := struct {
+		Version string `json:"version"`
```
Comment (Tomasvrba) [inline backend/bitboxbase/handlers/handlers.go]: If the frontend is providing the version, does that mean the user will have choices which version to update to?

Comment (0xB10C) [reply backend/bitboxbase/handlers/handlers.go]: Yes. That's how it's designed by @Stadicus, atleast for the MVP.

File: backend/bitboxbase/rpcmessages/rpcmessages.go (hunk 1 above)
Comment (0xB10C) [reply backend/bitboxbase/rpcmessages/rpcmessages.go]: Must have happend while coping from the `base/rpcmessages.go`.

**Comment by 0xB10C** [comment]
Addressed comments and rebased on master.

**Comment by 0xB10C** [comment]
As discussed offline: Since the rebase changed quite a few things, this needs manual testing via `curl` before merge by either @Tomasvrba or me.

**Comment by 0xB10C** [comment]
Closing this in favor of https://github.com/digitalbitbox/bitbox-wallet-app/pull/648 which is merged into `stageing`.

**Pull request closed without merging**